tests/test_aggregation.py

```diff
@@ -31,6 +31,8 @@ def test_schema_to_template_ds(specified_chunks):
             chunksize = var.chunksizes[dim]
             expected_chunksize = _expected_chunks(size, specified_chunks.get(dim, None))
             assert chunksize == expected_chunksize
+    # Confirm original time units have been preserved
+    assert ds.time.encoding["units"] == dst.time.encoding["units"]
```

Comment:
Would this test fail without the changes in `aggregation.py`?

Reply:
I checked and it would fail for the wrong reason (`KeyError`) in that situation, I've fixed it and will commit (also in `test_witers`)